src/nvim/window.c

```diff
@@ -1939,6 +1941,16 @@ int win_close(win_T *win, bool free_buf)
       return FAIL;
   }
 
+  // fire WinClosed event just before freeing memory associated with window
+  if (has_event(EVENT_WINCLOSED)) {
+    apply_autocmds(EVENT_WINCLOSED, win->w_buffer->b_fname,
+                   win->w_buffer->b_fname, false, win->w_buffer);
+
+    // If autocommands closed the window, there's nothing else to do
+    if (!win_valid(win)) {
+      return FAIL;
```

Comment:
Can you explain this? How could autocommands close the window during a `WinClosed` event for that very window?

Reply:
Yes, I'm trying to avoid recursion. But as I said before, I do not see where/how is `recursive` defined.